core/src/org/pentaho/di/core/Const.java

```diff
@@ -2420,7 +2444,13 @@ public static int indexOfString( String lookup, List<String> list ) {
             addSplit = oddNumberOfEnclosures;
           }
           if ( addSplit ) {
-            splitList.add( concatSplit.toString() );
+            String splitResult = concatSplit.toString();
+            //remove enclosure from resulting split
+            if ( removeEnclosure ) {
+              splitResult = splitResult.replaceAll( enclosure, "" );
```

Comment:
This doesn't seem to cover the case where splitResult includes the enclosure character in mid-string. Please review